stl/inc/atomic

```diff
@@ -1158,6 +1158,24 @@ struct _Atomic_integral<_Ty, 8> : _Atomic_storage<_Ty> { // atomic integral oper
 #endif // _M_IX86
 };
 
+#if _HAS_CXX17
+#if 1 // TRANSITION, ABI
+
+template <class _Ty>
+inline constexpr bool _Is_always_lock_free = sizeof(_Ty) <= 8 && (sizeof(_Ty) & (sizeof(_Ty) - 1)) == 0;
+
+#else // ^^^ don't break ABI / break ABI vvv
+
+#if _ATOMIC_HAS_DCAS
+template <class _Ty>
+inline constexpr bool _Is_always_lock_free = sizeof(_Ty) <= 2 * sizeof(void*);
+#else // ^^^ _ATOMIC_HAS_DCAS / !_ATOMIC_HAS_DCAS vvv
+template <class _Ty>
+inline constexpr bool _Is_always_lock_free = sizeof(_Ty) <= sizeof(void*);
+#endif // _ATOMIC_HAS_DCAS
+
+#endif // break ABI
+#endif // _HAS_CXX17

```

Comment:
I observe that `_Is_always_lock_free` cares only about the size of the type, and not the type itself, so we could give it a `size_t` non-type template parameter instead of a type template parameter and use `_Is_always_lock_free<sizeof(_Ty)>` at the callsites to reduce the total number of specializations. Should we do so?

Reply:
That is so much better, my hat is off